### ISSUE TITLE
`numeric` shouldn't allow null with BQ during translation

### ISSUE THREAD

**Issue by MichaelSchmidt82** [OHDSI/SqlRender#265]
Hi.

I'm using OHDSI/ETL-Synthea (v5.3.1-updates). During a call to `CreateVocabTables(...)` (which I don't expect you to know), the following SQL is loaded for translation when Google's BigQuery is specified: [here is the entire file](https://github.com/OHDSI/ETL-Synthea/blob/v5.3.1-updates/inst/sql/sql_server/create_vocab_tables.sql).

and here are the exact lines of code that are performed on the file mentioned above are:
```
renderedSql <- SqlRender::render(sqlQuery, vocab_schema = vocabDatabaseSchema)
translatedSql <- SqlRender::translate(renderedSql, targetDialect = connectionDetails$dbms)
```

and in particular, the following SQL is rendered/translated as part of the `translatedSql` output.  For brevity, the issue starts at  at line 102:
```
. . .
DROP TABLE IF EXISTS synthea_dummy_cdm5.drug_strength;
create table synthea_dummy_cdm5.drug_strength (
  drug_concept_id                      INT64          not null,
  ingredient_concept_id              INT64          not null,
  amount_value                          numeric            null,
  amount_unit_concept_id          INT64,
  numerator_value                      numeric            null,
  numerator_unit_concept_id        INT64,
  denominator_value                    numeric            null,
  denominator_unit_concept_id    INT64,
  box_size                                INT64,
  valid_start_date                    date            not null,
  valid_end_date                      date            not null,
  invalid_reason                      STRING
)
;
. . .
```

This is not correct which leads to a syntax error.  The problem is that fields listed as numeric can not be null.  I.E

```
 amount_value                          numeric            null,
 ```
 is wrong, whereas,
```
 amount_value                          numeric,
```
is valid.  So during SqlRender's translation, it should not allow numeric types to be null.  Removing the three instances of `null` allows the query to finish.

**Comment by schuemie** [comment]
Huh, I wonder why one would even specify `null`, as that seems to be implied when `not null` is not specified. I think we can come up with a translation rule that does this. Do you know if `numeric` is the only data type to which this applies? It would seem this applies to every data type on BQ?

**Comment by MichaelSchmidt82** [comment]
I'm actually unsure if this applies to every data type.  I can try other cases, but intuitively I believe this is the case for any data type given the logic above.  I will try some other data types and see if this is the case.

**Comment by MaximMoinat** [comment]
Same seems to happen for at least the `datetime` type (see linked issue ^)

As far as I can see, the `NULL` syntax is not supported by BQ. And in SqlRender, there are already replacement patterns for the `string` and `int64` type to remove the NULL:
https://github.com/OHDSI/SqlRender/blob/a3b38a9306c651b005718acf1aaaef35d7b56d50/inst/csv/replacementPatterns.csv#L679-L680

So simply adding these replacement patters for all BQ data types might fix this issue.

**Comment by MichaelSchmidt82** [comment]
Has anyone had a chance to review this?

**Comment by schuemie** [comment]
Will be fixed in imminent release (1.9.1)
